plutus-tx/src/PlutusTx/Enum.hs

```diff
@@ -0,0 +1,17 @@
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+module PlutusTx.Enum (Enum(..)) where
+
+import           PlutusTx.Builtins
+import           Prelude           hiding (Enum (..))
+
+class Enum a where
+  succ :: a -> a
+  pred :: a -> a
+  toEnum :: Integer -> a
+  fromEnum :: a -> Integer
+
+instance Enum Integer where
```

Comment:
This could do with some more instances, `Bool`, `Maybe`, `()`, etc.

Reply:
Added instances for `Bool`, `Ordering` and `()`.